### PR TITLE
[Infer Symbolic Shape No.245,247] Add symbol_infer_interface for weight_{dequantize.quantize}

### PR DESCRIPTION
**Pull request by ooooo-create**
### PR Category

CINN

### PR Types

Others

### Description

- Add symbol_infer_interface for weight_dequantize
- Add symbol_infer_interface for weight_quantize

### CODE REVIEW

**Review by gongshaotian — APPROVED** [review]
LGTM

File: test/quantization/test_weight_only_linear.py
Comment (gongshaotian) [inline test/quantization/test_weight_only_linear.py:1]: 功能测试完之后可以回退

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/binary_infer_sym.cc [hunk 1]
```diff
@@ -2163,12 +2163,60 @@ bool UnpoolOpInferSymbolicShape(pir::Operation *op,
...
+      infer_context->GetShapeOrDataForValue(op->operand_source(0)).shape();
+  const auto &scale_shape =
+      infer_context->GetShapeOrDataForValue(op->operand_source(1)).shape();
+  PADDLE_ENFORCE_EQ(x_shape.size(),
+                    2UL,
+                    common::errors::InvalidArgument(
+                        "The x tensor of dequantize op must be 2D, but got[%d]",
+                        x_shape.size()));
+  int group_size = op->attribute<pir::Int32Attribute>("group_size").data();
+  std::string algo = op->attribute<pir::StrAttribute>("algo").AsString();
```
Comment (zyfncg) [inline paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/binary_infer_sym.cc:2178]: 不变的变量最好都加上const前缀

**Review by zyfncg — APPROVED** [review]

File: paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/binary_infer_sym.cc (hunk 1 above)
Comment (ooooo-create) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/binary_infer_sym.cc:2178]: 好的，收到

Comment (gongshaotian) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/binary_infer_sym.cc:2178]: const 可以放后面pr中修改

Comment (ooooo-create) [reply paddle/fluid/pir/dialect/operator/interface/infer_symbolic_shape/binary_infer_sym.cc:2178]: 好的，下个pr进行修改

**Pull request merged**